File: programmers/algorithm/level2/hellozo0/week23/타겟넘버.java
package programmers.algorithm.level2.hellozo0.week23;

public class 타겟넘버 {
    class Solution {

        static int answer;

        public int solution(int[] numbers, int target) {
            answer = 0;
            DFS(numbers,target, 0,0);

            return answer;
        }

        public static void DFS(int[] numbers, int target, int index, int sum) {
            if(index == numbers.length){
                if(sum == target) answer++;
            } else {
                DFS(numbers,target,index+1, sum + numbers[index]);
                DFS(numbers,target,index+1, sum - numbers[index]);
            }

        }
    }
}

File: programmers/algorithm/level2/Hong0329/타겟 넘버.cpp
//
// Created by hong jun chung on 8/11/24.
//
#include <string>
#include <vector>

using namespace std;

int answer = 0;

void dfs(vector<int> numbers, int target, int sum, int num) {
    if(num == numbers.size()) {
        if(sum == target) {
            answer++;
        }
        return;
    }
    dfs(numbers, target, sum + numbers[num], num + 1);
    dfs(numbers, target, sum - numbers[num], num + 1);
}

int solution(vector<int> numbers, int target) {
    dfs(numbers, target, 0, 0);
    return answer;
}